app/src/test/java/com/evercheck/flagly/domain/usecase/SetFeatureFlagUseCaseTest.kt

```diff
@@ -0,0 +1,62 @@
+package com.evercheck.flagly.domain.usecase
+
+import com.evercheck.flagly.data.DynamicFeatureFlagHandler
+import com.evercheck.flagly.domain.model.FeatureFlag
+import io.mockk.mockk
+import io.mockk.verify
+import java.util.Random
+import org.junit.Before
+import org.junit.Test
+
+class SetFeatureFlagUseCaseTest {
+
+    private val localFeatureFlagHandler = mockk<DynamicFeatureFlagHandler>(relaxed = true)
+
+    private lateinit var setFeatureFlagUseCase: SetFeatureFlagUseCase
+
+    @Before
+    fun setup() {
+        setFeatureFlagUseCase = SetFeatureFlagUseCase(localFeatureFlagHandler)
+    }
+
+    private val featureFlag = object : FeatureFlag {
+        override val name: String
+            get() = "one"
+    }
+
+    @Test
+    fun `valid with the value from override is true or false`() {
+        val numberRandom = Random().nextInt(2)
+        val value = numberRandom % 2 == 0
+
+        setFeatureFlagUseCase(featureFlag, value, remoteValue = true)
+
+        if (value) {
+            verify(exactly = 1) {
+                localFeatureFlagHandler.setValue(featureFlag, value)
+            }
+        } else {
+            verify(exactly = 1) {
+                localFeatureFlagHandler.removeOverridenValue(featureFlag)
+            }
```

Comment:
you should not have an if else into a test, create different test for this purposes